modules/camera/camera_device.py

```diff
@@ -8,24 +8,33 @@
 import cv2
 import numpy as np
 
+import picamera2
+
 
 class CameraDevice:
     """
     Wrapper for camera.
     """
 
-    def __init__(self, name: "int | str", save_nth_image: int = 0, save_name: str = "") -> None:
+    def __init__(
+        self, name: "int | str", save_nth_image: int = 0, save_name: str = "", use_pc2: bool = False
+    ) -> None:
         """
         name: Device name or index (e.g. /dev/video0 ).
         (optional) save_nth_image: For debugging, saves every nth image.
             A value of 0 indicates no images should be saved
         (optional) save_name: For debugging, file name for saved images.
+        (optional) use_pc2: Use picamera2 implementation instead of opencv impl
         """
-        self.__camera = cv2.VideoCapture(name)
-        assert self.__camera.isOpened()
-
-        self.__camera.set(cv2.CAP_PROP_FRAME_WIDTH, sys.maxsize)
-        self.__camera.set(cv2.CAP_PROP_FRAME_HEIGHT, sys.maxsize)
+        self.__using_pc2 = use_pc2
+        if self.__using_pc2:
+            self.__camera = picamera2.Picamera2()
```

Comment:
Set explicit resolution (pass it in as an input) (see 4.2.2.1)
Also, you may need to change the colour space of the picamera, it may not use RGB by default (see 4.2.2.2)

Reply:
Ah okay, I noticed this issue when researching online. So we want to use RGB?